--- magento/module-catalog/Pricing/Price/TierPrice.php
+++ magento/module-catalog/Pricing/Price/TierPrice.php
@@ -177,7 +177,7 @@ function (&$priceData) {
                     /* convert string value to float */
                     $priceData['price_qty'] *= 1;
                     if ($this->getConfigTaxDisplayType() === Config::DISPLAY_TYPE_BOTH) {
-                        $exclTaxPrice = $this->calculator->getAmount($priceData['price'], $this->product, true);
+                        $exclTaxPrice = $this->calculator->getAmount($priceData['price'], $this->product);
                         $priceData['excl_tax_price'] = $exclTaxPrice;
                     }
                     $priceData['price'] = $this->applyAdjustment($priceData['price']);
