diff --git a/Model/Label/BlockRepository.php b/Model/Label/BlockRepository.php
index 6f9172d..1bcf186 100644
--- a/Model/Label/BlockRepository.php
+++ b/Model/Label/BlockRepository.php
@@ -58,7 +58,7 @@ class BlockRepository implements BlockRepositoryInterface

         foreach ($availableRules as $availableRule) {
             $label = $this->findLabelById($labels, $availableRule[RuleProductInterface::LABEL_ID]);
-            if (!$this->counter->isLimitReached($label->getPosition(), $storeId)) {
+            if ($label && !$this->counter->isLimitReached($label->getPosition(), $storeId)) {
                 $labelTexts = [
                     RuleProductInterface::LABEL_TEXT_LARGE => $availableRule[RuleProductInterface::LABEL_TEXT_LARGE],
                     RuleProductInterface::LABEL_TEXT_MEDIUM => $availableRule[RuleProductInterface::LABEL_TEXT_MEDIUM],
