diff --git a/Model/ProductPermission/Resolver.php b/Model/ProductPermission/Resolver.php
index e83ea8d91..306ba5262 100644
--- a/Model/ProductPermission/Resolver.php
+++ b/Model/ProductPermission/Resolver.php
@@ -1,4 +1,19 @@
 <?php
+/**
+ * Aheadworks Inc.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the EULA
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://aheadworks.com/end-user-license-agreement/
+ *
+ * @package    CustGroupCatPermissions
+ * @version    1.0.8
+ * @copyright  Copyright (c) 2021 Aheadworks Inc. (https://aheadworks.com/)
+ * @license    https://aheadworks.com/end-user-license-agreement/
+ */
 namespace Aheadworks\CustGroupCatPermissions\Model\ProductPermission;
 
 use Aheadworks\CustGroupCatPermissions\Api\Data\CategoryPermissionInterface;
@@ -108,12 +123,16 @@ class Resolver extends AbstractPermissionResolver
      */
     protected function getPermissionByCategories($categories)
     {
-        foreach ($categories as $category) {
-            $result = $this->categoryPermissionResolver->getPermissionForCategory($category);
+        if ($categories) {
+            foreach ($categories as $category) {
+                $result = $this->categoryPermissionResolver->getPermissionForCategory($category);
 
-            if ($result->getViewMode() == AccessMode::HIDE || $result->getPriceMode() == AccessMode::HIDE) {
-                break;
+                if ($result->getViewMode() == AccessMode::HIDE || $result->getPriceMode() == AccessMode::HIDE) {
+                    break;
+                }
             }
+        } else {
+            $result = $this->getDefaultPermission();
         }
 
         return $result;
@@ -127,3 +146,4 @@ class Resolver extends AbstractPermissionResolver
         return $this->permissionConverter->getDataObject($permissionData);
     }
 }
+
